### PR TITLE
Cardigann: add check for redirect when pulling login page

### PR DESCRIPTION
**Pull request by abeloin**
As per @ilike2burnthing [request](https://github.com/Jackett/Jackett/issues/9029#issuecomment-716907578), here's a PR that add a check for redirect when pulling the login page.

This fix the redirect(307) issue with yggtorrent. See https://github.com/Jackett/Jackett/issues/9959#issuecomment-716058618 for more info.

It was not tested with other indexers.

### CODE REVIEW

**Comment by ngosang** [comment]
The code LGTM, just add a comment in the code with the issue you are solving.
@garfield69 I don't know what to do with this PR. It can cause issues in other trackers and we can't be sure... You can merge it and see what happens...

**Comment by abeloin** [comment]
Comment added and changed the if check to limit it to ygg only via Definition.Id.

This should limit the impact to other tracker to a minimum.

File: src/Jackett.Common/Indexers/CardigannIndexer.cs [hunk 1]
```diff
@@ -880,7 +880,9 @@ public async Task<ConfigurationData> GetConfigurationForSetup(bool automaticlogi
                 configData.CookieHeader.Value = string.Join("; ", Login.Cookies);
             landingResult = await RequestWithCookiesAsync(LoginUrl.AbsoluteUri, referer: SiteLink);
 
-            if (landingResult.IsRedirect)
+            // Some sites have a temporary redirect before the login page, we need to process it.
+            // This check only apply to sites requiring it, which is currently: Yggtorrent
+            if (Definition.Id == "yggtorrent" && landingResult.IsRedirect)
```
Comment (garfield69) [inline src/Jackett.Common/Indexers/CardigannIndexer.cs:885]: IMO a check that the 307 is occurring for just yggtorrent is the wrong approach.

any yml indexer that has the **followredirect: true**  statement 
https://github.com/Jackett/Jackett/blob/4df521562841e4142ee9fa0cb859db1fa71f1890/src/Jackett.Common/Definitions/yggtorrent.yml#L8
should allow for the 307 redirect to be accepted and processed.

this code just needs to confirm that the followredirect is enabled and then handle a 307 or reject it if not allowed.

**Comment by ilike2burnthing** [comment]
Tested with all public trackers and most semi-private, no issues. Working for YGGtorrent.

**Comment by abeloin** [comment]
@garfield69 Thanks for the feedback.

I've updated the code to check for "Followredirect" and removed "landingResult.IsRedirect" since it's the first thing done in "FollowIfRedirect()" loop.

If the code looks good, let me know and I'll squash the commits into one.

**Review by garfield69 — APPROVED** [review]
Tested, LGTM.

**Comment by garfield69** [comment]
Jackett 0.16.1922

**Comment by julian-ladjani** [comment]
I think that this problem is not fixed #10842

**Pull request merged**